### ISSUE TITLE
`/context`, `/messages`, and `/sync` state

### ISSUE THREAD

**Issue by tezlm** [matrix-org/matrix-spec#1641]
**Link to problem area**:

[/rooms/{roomId}/context/{eventId}](https://spec.matrix.org/v1.8/client-server-api/#get_matrixclientv3roomsroomidcontexteventid)

[/rooms/{roomId}/messages](https://spec.matrix.org/v1.8/client-server-api/#get_matrixclientv3roomsroomidmessages)

[/sync](https://spec.matrix.org/v1.8/client-server-api/#get_matrixclientv3sync)

**Issue**

- `state` in `/context` is the state of the room at the last event returned
- `state` in `/sync` is start of the returned timeline (first event returned) 
- `state` in `/messages` seems to be undefined?

These are all different, and `/messages` is undefined?

**Comment by richvdh** [comment]
yes they are all different. Unlike the other two, the state in `/messages` is not the full room state, just the membership events for messages in the returned chunk when lazy-loading is enabled.

Questions about the protocol are better directed to `#matrix-spec:matrix.org`

**Comment by tezlm** [comment]
in that case, it might be worth clarifying the spec to mention that `/messages` state is for lazy loaded member events

**Comment by richvdh** [comment]
> For example, if `lazy_load_members` is enabled in the filter then this may contain the membership events for the senders of events in the chunk.
